src/app/oauth2/middleware.js

```diff
@@ -28,26 +21,18 @@ module.exports = {
 
   initSessionWithJWT: async (req, res, next) => {
     const requestJWT = req.jwt;
-    const headers = { client_id: req.query?.client_id };
-
     try {
       if (requestJWT) {
-        const apiResponse = await req.axios.post(
-          AUTHORIZE,
-          {
-            request: req.jwt,
-            ...req.session.authParams,
-          },
-          {
-            headers: headers,
-          }
-        );
+        const apiResponse = await req.axios.post(AUTHORIZE, {
+          request: req.jwt,
+          client_id: req.session.authParams.client_id,
+        });
         req.session.tokenId = apiResponse?.data["session_id"];
       }
+      next();
```

Comment:
I think I very slightly prefer a tail-next as opposed to next inserted in the flow - but that isn't very important.

What I think is important, is that all of the next invocations should be `return next()` - that is a definite fix for the `headers already sent` issue.

If we cared enough about testing for that, we could add a `expect(next).to.have.only.been.called.once` test somewhere.

Reply:
I'll change this one here and then raise another PR with changes to other instances of it